src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -70,39 +69,69 @@ case object ThurloeDatabaseConnector extends DataAccess {
   def collectAll(userId: String): Future[UserKeyValuePairs] = {
     for {
       results <- lookupWithConstraint(x => x.userId === userId)
-    } yield UserKeyValuePairs(userId, results)
+    } yield UserKeyValuePairs(userId, results map { x => x.keyValuePair })
   }
 
+  import thurloe.database.DatabaseOperation.DatabaseOperation
   /**
    * Writes the user key value pair to the database, with the exception of the 'value' which uses
    * the encrypted version instead. The IV also comes from the encryption result.
    */
-  private def databaseWrite(userKeyValuePair: UserKeyValuePair, encryptedValue: EncryptedBytes): Future[Unit] = {
-    val keyValuePairs = TableQuery[DbKeyValuePair]
+  private def databaseWrite(userKeyValuePair: UserKeyValuePair, encryptedValue: EncryptedBytes): Future[SetKeyResponse] = {
 
-    val action = keyValuePairs.insertOrUpdate(
-      None,
-      userKeyValuePair.userId,
-      userKeyValuePair.keyValuePair.key,
-      encryptedValue.base64CipherText,
-      encryptedValue.base64Iv
-    )
+    val lookupExists = keyLookup(userKeyValuePair.userId, userKeyValuePair.keyValuePair.key)
```

Comment:
any reason not to simplify these method names to `lookup`, `insert`, and `update`?